amgut/lib/data_access/ag_data_access.py

```diff
@@ -386,6 +386,58 @@ def updateAGBarcode(self, barcode, ag_kit_id, site_sampled,
                                     refunded, withdrawn])
         self.connection.commit()
 
+    def registerHandoutKit(self, ag_login_id, supplied_kit_id):
+        """
+        Returns
+        -------
+        bool
+            True:  success
+            False: insert failed due to IntegrityError
+
+        Notes
+        -----
+        Whatever is passed as kit_password will be added AS IS. This means you
+        must hash the password before passing, if desired.
+        """
+        printresults = self.checkPrintResults(supplied_kit_id)
+        if printresults is None:
+            printresults = 'n'
+
+        sql = """
+            DO $do$
+            DECLARE
+                k_id varchar;
+                bc varchar;
+            BEGIN
+                INSERT INTO ag_kit
+                (ag_login_id, supplied_kit_id, kit_password, swabs_per_kit,
+                 kit_verification_code, print_results)
+                SELECT '{0}', '{1}', password, swabs_per_kit,
+                    verification_code, '{2}'
+                    FROM ag_handout_kits WHERE kit_id = '{1}'
+                RETURNING ag_kit_id INTO k_id;
+                FOR bc IN
+                    SELECT barcode FROM ag_handout_kits WHERE kit_id = '{1}'
+                LOOP
+                    INSERT INTO barcode (barcode) VALUES (bc);
+                    INSERT INTO project_barcode (project_id, barcode)
+                        VALUES (1, 'bc');
+                    INSERT  INTO ag_kit_barcodes
+                        (ag_kit_id, barcode, sample_barcode_file)
+                        VALUES (k_id, 'bc', 'bc' || '.jpg');
+                END LOOP;
+                DELETE FROM ag_handout_kits WHERE kit_id = '{1}';
+            END $do$;
+            """.format(ag_login_id, supplied_kit_id, printresults)
```

Comment:
`supplied_kit_id` isn't scrubbed here, so I believe this is prone to an injection. Possible to pass this in during the execute call and let psycopg2 take care of the scrubbing?